pkgs/by-name/de/decasify/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+
+  # nativeBuildInputs
+  zstd,
+  pkg-config,
+  jq,
+  cargo,
+  rustc,
+  rustPlatform,
+
+  # buildInputs
+  typos,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "decasify";
+  version = "0.8.0";
+
+  src = fetchurl {
+    url = "https://github.com/alerque/decasify/releases/download/v${finalAttrs.version}/decasify-${finalAttrs.version}.tar.zst";
+    sha256 = "sha256-HTUAb/yL3H4B/n/Ecd/fDpnTYiqwco/E07sa6pFIIU4=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
```

Comment:
```suggestion
    inherit (finalAttrs) pname version src;
```

Reply:
This change causes the build to fail.

Reply:
@alerque adding `pname` and `version` there has the small benefit of a small benefit of producing a slightly more informative `$out` path:

```
/nix/store/x2l73974ganpvd79sdwfl0b5z1gjvlkk-decasify-0.8.0-vendor.tar.gz
```

Instead of:

```
/nix/store/c5lj5pl5ladzb9yw6mg6gkabh8xzyh6r-cargo-deps-vendor.tar.gz
```

The unfortunate detail that made this change "fail" for you, is the fact that this causes a change of hash, see:

- https://github.com/NixOS/nixpkgs/issues/85703

This worked for me locally:

```nix
  cargoDeps = rustPlatform.fetchCargoTarball {
    inherit (finalAttrs) pname version src;
    nativeBuildInputs = [ zstd ];
    dontConfigure = true;
    hash = "sha256-bD8MYufI87j//7dIAnCzmp4yoOaT81Zv1i7rjWpjPlc=";
  };
```

Reply:
Whose idea was it to make the final filename part of the hash (but only part of the file name because of course the hash itself is also in the file name)? On second thought maybe don't tell me, I might be inclined to take `git blame` literally ;-P